### PR TITLE
MODSOURMAN-146 - Move code to the data-import-utils

### PR DESCRIPTION
**Pull request by RuslanLavrov**

### CODE REVIEW

File: src/main/java/org/folio/dataimport/util/Try.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+public class Try {
+
+  private Try() {
+  }
+
+  /**
+   * Return future with result returned by specified task.
+   * If the {@code task} throws an exception, the returned future will be failed with this exception
+   *
```
Comment (Anatolii-Starkov) [inline src/main/java/org/folio/dataimport/util/Try.java]: I sugest:
Executes a task and returns either a future with result or a future with an exception

File: src/main/java/org/folio/dataimport/util/Try.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (Exception e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  /**
+   * Return succeed or failed future handled by specified job.
+   * If the {@code job} throws an exception, the returned future will be failed with this exception
+   *
```
Comment (Anatolii-Starkov) [inline src/main/java/org/folio/dataimport/util/Try.java]: I sugest:
Executes a job. It creates new future and pass it to a job as a parameter, so you are able to control it. If a job is failed then the future is completed and propagated with an exception.

File: src/test/java/org/folio/dataimport/util/TryTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.runners.JUnit4;
+
+import java.util.function.Supplier;
+
+@RunWith(JUnit4.class)
+public class TryTest {
+
+  @Test
+  public void shouldReturnSucceededFutureWhenTaskReturnedResult() {
+    Supplier<String> task = () -> "result string";
```
Comment (Anatolii-Starkov) [inline src/test/java/org/folio/dataimport/util/TryTest.java]: What is the main reason to extract the task as a local variable?

- It's very short

- It is not express  the main Idea of the Try calss

- Extra line of code

Comment (Anatolii-Starkov) [reply src/test/java/org/folio/dataimport/util/TryTest.java]: I played around it and rewrite it a little
```
public class TryTest {

  private static final String RESULT_STRING = "result string";

  @Test
  public void shouldReturnSucceededFutureWhenTaskReturnedResult() {
    Try.itGet(() -> RESULT_STRING)
      .setHandler(ar -> {
        Assert.assertTrue(ar.succeeded());
        Assert.assertEquals(RESULT_STRING, ar.result());
      });
  }

  @Test
  public void shouldReturnFailedFutureWhenTaskThrewException() {

    Try.itGet(this::throwException)
      .setHandler(ar -> {
        Assert.assertTrue(ar.failed());
        Assert.assertThat(ar.cause(), instanceOf(RuntimeException.class));
      });
  }

  @Test
  public void shouldReturnSucceededFutureWhenJobReturnedResult() {
    Try.itDo(future -> future.complete(RESULT_STRING))
      .setHandler(ar -> {
        Assert.assertTrue(ar.succeeded());
        Assert.assertEquals(RESULT_STRING, ar.result());
      });
  }

  @Test
  public void shouldReturnFailedFutureWhenJobThrewException() {

    Try.itDo(future -> throwException())
      .setHandler(ar -> {
        Assert.assertTrue(ar.failed());
        Assert.assertThat(ar.cause(), instanceOf(RuntimeException.class));
      });
  }

  private String throwException() {
    throw new RuntimeException();
  }
}
```

File: src/main/java/org/folio/dataimport/util/Try.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      job.accept(future);
+    } catch (Exception e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  @FunctionalInterface
+  public interface Job<T> {
+    void accept(T t) throws Exception;//NOSONAR
```
Comment (Anatolii-Starkov) [inline src/main/java/org/folio/dataimport/util/Try.java]: I suggest:
Performs this operation on the given argument, or throws an exception if unable to do so.

File: src/main/java/org/folio/dataimport/util/Try.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+      job.accept(future);
+    } catch (Exception e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  @FunctionalInterface
+  public interface Job<T> {
```
Comment (Anatolii-Starkov) [inline src/main/java/org/folio/dataimport/util/Try.java]: I sugest: 
Represents an operation that accepts a single input argument and returns no result  or throws an exception if unable to do so.

File: src/main/java/org/folio/dataimport/util/RestUtil.java [hunk 6]
```diff
@@ -181,4 +181,35 @@ private static boolean isJson(AsyncResult<WrappedResponse> asyncResult) {
...
+  }
+
+  /**
+   * Checks whether response content type is "application/json"
+   *
+   * @param response - http response
+   * @return - true if response content type is "application/json", otherwise false
+   */
+  public static boolean isContentTypeJson(HttpClientResponse response) {
+    return APPLICATION_JSON.equals(response.getHeader(CONTENT_TYPE));
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/dataimport/util/RestUtil.java]: Do we really need this method ?

File: src/main/java/org/folio/dataimport/util/RestUtil.java [hunk 7]
```diff
@@ -181,4 +181,35 @@ private static boolean isJson(AsyncResult<WrappedResponse> asyncResult) {
     HttpClientResponse response = asyncResult.result().response;
     return response != null && APPLICATION_JSON.equals(response.getHeader(CONTENT_TYPE));
   }
+
+  /**
+   * Checks whether the answer is partial successful
+   *
+   * @param response - http response
+   * @return - true if response status is "server error" and content type is "application/json", otherwise false
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/dataimport/util/RestUtil.java]: just a note - we better not forget to get rid of this method once partial success status is changed, maybe wee should create a ticket for that?

Comment (RuslanLavrov) [reply src/main/java/org/folio/dataimport/util/RestUtil.java]: reference to created task for this issue [MODDATAIMP-150](https://issues.folio.org/browse/MODDATAIMP-150)

**Pull request merged**